### ISSUE TITLE
Gracefully Handle WAF Rule Deletions

### ISSUE THREAD

**Issue by Evesy** [terraform-providers/terraform-provider-cloudflare#535]
We're explicitly defining all Cloudflare managed WAF rules in Terraform. Occasionally Cloudflare will remove a WAF rule (e.g. merging what was a beta rule into the main rule).

We are notified via RSS feed (https://developers.cloudflare.com/waf/change-log/) when rules are changed (either removed, added, or default behaviour changed).

When a rule is removed, we'll reflect this change in our Terraform by removing the rule from our list. However, since the rule no longer exists, Terraform fails at the plan stage as it cannot get the state of that resource from Cloudflare API, as the rule ID no longer exists.

In this instance, we have to first manually delete the resource from the Terraform state. This could be 5 or 6 rules, across multiple zones, which isn't ideal.

### Terraform Version
0.12.12

### Affected Resource(s)
Please list the resources as a list, for example:
- cloudflare_waf_rule

### Terraform Configuration Files
```hcl
resource "cloudflare_waf_rule" "cfwaf" {
  for_each = var.cloudflare_managed_rules

  zone_id = cloudflare_zone.default.id
  rule_id   = "100044_BETA"
  mode    = "simulate"
}
```

### Expected Behavior
Not exactly sure this is 'unexpected' behaviour, but it'd be nice to be able to remove these resources that no longer exist in the Cloudflare API without having to manually manipulate state

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Add waf rule to Terraform
2. After rule is removed by Cloudflare, attempt to plan Terraform again

**Comment by jacobbednarz** [comment]
The only sort of stop gap solution I could think of is to check whether the API returns a 404 around https://github.com/terraform-providers/terraform-provider-cloudflare/blob/b5559383e9c6b0b72d5ec6160617270d7d876888/cloudflare/resource_cloudflare_waf_rule.go#L54 and if so, remove it from the Terraform state. There are some caveats to this approach however I don't think there are any other options. Here is an example of this approach using `cloudflare_load_balancer_monitor` https://github.com/terraform-providers/terraform-provider-cloudflare/blob/7df730622f105a6737afedd2c24453c855b32322/cloudflare/resource_cloudflare_load_balancer_monitor.go#L312-L315

**Comment by jacobbednarz** [comment]
This should have been fixed in #623. Please ping me back if it hasn't.
